corgidrp/l4_to_tda.py

```diff
@@ -0,0 +1,73 @@
+# A file that holds the functions that transmogrify l4 data to TDA (Technical Demo Analysis) data 
+import corgidrp.fluxcal as fluxcal
+
+def determine_color_cor(input_dataset, calspec_filepath, source_filepath):
+    """
+    determine the color correction factor of the observed source
+    at the reference wavelength of the used filter band and put it into the header.
+    We assume that each frame in the dataset was observed with the same color filter.
```

Comment:
@rzellem - up to now I assume that all frames in a dataset use the same color filter and get it from the header keyword "CFAMNAME"...otherwise we must sort the dataset accordingly.

Reply:
@JuergenSchreiber - thanks! I'm just being a bit paranoid that we might have occasionally user or pipeline error and data will get mixed in...but maybe I'm just being too paranoid here....

Reply:
instead of updating via the history entry, you can just update the headers in a for loop. that way (I assume it's not computationally intensive to compute the color correction, so it's ok if we repeat it a bunch of times).

Reply:
My naive assumption was, that we have an input dataset of the source star in a defined color filter band, and update each frame header with the same K-factor.  If we would have different star observations e.g. in different filters in one dataset, we would indeed need to run this step for each single frame. But I think we only have one observation in one dataset, isn't it?

Reply:
For images taken without the ND filter, we have only one exposure per star. However, for images taken *with* the ND filter, we will do a pointing grid pattern of multiple exposures - we need to do this to ensure that we align the star with the ND filter's sweet spot (as there's some pointing instabilities that will prevent us from directly aligning with the sweet spot) - so we will select only 1 of the bunch that best aligns with the sweet spot.

Reply:
I'm not sure we know all the possible ways we want to use the pipeline for flux calibration (e.g., processing multiple stars together seems unlikely, but you never know). I'm fine punting this until later. I think putting this in a for loop later is not a problem if we determine that's the workflow we want to do.